Sources/APILayer/Service/ApiService.swift

```diff
@@ -167,23 +167,24 @@ public extension ApiService {
     }
 }
 
-///Manage uploading files
+/// Manage uploading files
 public extension ApiService {
 
     /**
      Uploads file using HTTP POST request.
 
      - Parameters:
-       - localFileUrl: URL on disc of the resource to upload.
-       - destinationUrl: URL of the receiver.
-       - aditionalHeaders: Array of all aditional HTTP header fields.
-       - inBackground: Flag indicates if uploading should be performed in background or foreground.
-       - useProgress: Flag indicates if Progress object should be created.
-       - completion: Closure called when request is finished.
+         - localFileUrl: URL on disc of the resource to upload.
+         - destinationUrl: URL of the receiver.
+         - aditionalHeaders: Array of all aditional HTTP header fields.
+         - inBackground: Flag indicates if uploading should be performed in background or foreground.
+         - useProgress: Flag indicates if Progress object should be created.
+         - configuration: One of predefined *ApiService.Configuration* object containing request configuration.
+         - completion: Closure called when request is finished.
 
      - Returns: ApiRequest object which allows to follow progress and manage request.
      */
-    func postFile(from localFileUrl: URL, to destinationUrl: URL, with aditionalHeaders: [ApiHeader]? = nil, inBackground: Bool = true, useProgress: Bool = true, completion: ApiResponseCompletionHandler? = nil) -> ApiRequest {
+    func postFile(from localFileUrl: URL, to destinationUrl: URL, with aditionalHeaders: [ApiHeader]? = nil, inBackground: Bool = true, useProgress: Bool = true, configuration: ApiService.Configuration = ApiService.Configuration.foreground, completion: ApiResponseCompletionHandler? = nil) -> ApiRequest {
         let configuration: Configuration = inBackground ? .background : .foreground
```

Comment:
Without removing this line, configuration from method parameter won't be used 😉.
Also inserting configuration as a parameter IMO `inBackground` is no longer needed.

Reply:
That is true! In real i didn't dig that deep, it was just mechanical expansion of every method that had something to do with ApiRequest. Thank you for noticing! Hours of debugging in the future saved! 🥇 

The parameter `inBackground`  indeed is not needed when combined with the other